source/common/stats/allocator_impl.cc

```diff
@@ -256,6 +282,19 @@ GaugeSharedPtr AllocatorImpl::makeGauge(StatName name, StatName tag_extracted_na
   return gauge;
 }
 
+TextReadoutSharedPtr AllocatorImpl::makeTextReadout(StatName name, StatName tag_extracted_name,
+                                                    const StatNameTagVector& stat_name_tags) {
+  Thread::LockGuard lock(mutex_);
+  ASSERT(gauges_.find(name) == gauges_.end());
```

Comment:
also assert there's no counter of the same name? And in the makeCounter and makGauge impls make sure there are no TextReadouts of the same name.